tests/stdlib/talgorithm.nim

```diff
@@ -113,3 +113,152 @@ block:
     doAssert binarySearch(moreData, 6) == -1
     doAssert binarySearch(moreData, 4711) == 4
     doAssert binarySearch(moreData, 4712) == -1
+
+# merge
+block:
```

Comment:
can you try the usual `template main` + static tests?

Reply:
another PR of mine also change this test, so I only address half issues.